src/Main.hs

```diff
@@ -120,12 +126,11 @@ lookupDatatype :: Text -> PreMonad ()
 lookupDatatype ty = do
   traceM $ "lookupDatatype: " <> show ty
   (dm, s) <- get
-  let mTy = M.lookup ty dm
   case M.lookup ty dm of
-    Nothing -> if S.member ty s then pure () else do
+    Nothing -> if S.member ty s then pure () else
       error $ "can't look up datatype: " <> show ty
     Just typ -> case references typ of
-      [] -> proceed ty typ >> sortDatatypesByDeps
+      [] -> proceed ty typ -- >> sortDatatypesByDeps -- Pretty sure this was a bug
```

Comment:
I agree, `sortDatatypesByDeps` looks unnecessary here. Could you please just remove it, there is no point in keeping it commented out.